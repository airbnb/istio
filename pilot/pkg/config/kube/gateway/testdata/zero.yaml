apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  gatewayClassName: istio
  listeners:
  - hostname: "*.domain.example"
    port: 80
    protocol: HTTP
    routes:
      namespaces:
        from: All
      kind: HTTPRoute
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  namespace: default
spec:
  gateways:
    allow: All
  hostnames: ["first.domain.example"]
  rules:
  - matches:
    - path:
        type: Prefix
        value: /get
    forwardTo:
      - serviceName: httpbin-zero
        port: 8080
        weight: 0
  - matches:
    - path:
        type: Prefix
        value: /weighted-100
    forwardTo:
    - filters:
      - requestHeaderModifier:
          add:
            foo: bar
        type: RequestHeaderModifier
      port: 8000
      serviceName: foo-svc
      weight: 100
