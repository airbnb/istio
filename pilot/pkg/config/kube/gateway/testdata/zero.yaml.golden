apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*.domain.example'
    port:
      name: 0-gateway-gateway-istio-system
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - first.domain.example
  http:
  - match:
    - uri:
        prefix: /get
    route:
    - destination:
        host: httpbin-zero.default.svc.domain.suffix
        port:
          number: 8080
      weight: 0
    fault:
      abort:
        percentage:
          value: 100
        httpStatus: 503
  - match:
    - uri:
        prefix: /weighted-100
    route:
    - destination:
        host: foo-svc.default.svc.domain.suffix
        port:
          number: 8000
      headers:
        request:
          add:
            foo: bar
---
